src/coreclr/jit/importer.cpp

```diff
@@ -8989,6 +8989,7 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                         FALLTHROUGH;
                     case CORINFO_FIELD_STATIC_SHARED_STATIC_HELPER:
                     case CORINFO_FIELD_STATIC_ADDRESS:
+                    case CORINFO_FIELD_STATIC_RELOCATABLE:
```

Comment:
@EgorBo could you have a look at this? This affects a lot more than just RuntimeType and I might be missing why we weren't doing this before. (We might also want to do this for `STATIC_RVA_ADDRESS`.)

Reply:
@MichalStrehovsky wow, it's actually a good catch! It should improve some things, e.g.:
```cs
static readonly string Fld = "Hello";

[MethodImpl(MethodImplOptions.NoInlining)]
static string Test() => Fld;
```
```diff
; Assembly listing for method Benchmarks:Test():System.String (FullOpts)
-      mov      rax, qword ptr [(reloc 0x400000000042c388)]
-      mov      rax, gword ptr [rax+0x08]
+      lea      rax, gword ptr [(reloc 0x400000000042c398)]      ; '"Hello"'
       ret      
; Total bytes of code 12
```

Reply:
Basically, `static readonly` wasn't foldable for gc types for NAOT